lib/logstash/outputs/scalyr.rb

```diff
@@ -930,6 +937,12 @@ def build_multi_event_request_array(logstash_events)
           end
         end
 
+        if event_json.bytesize > @max_record_size_bytes
```

Comment:
Just thinking out loud - wonder what kind of performance impact this may have during the worst case (a lot of events which hit this limit).

Since previously we wouldn't perform additional processing in such case on the client side, right (aka we would send it as-is and server would reject / truncate it)?

May be worthwhile to add some instrumentation / period logging (number of such events, duration of truncation operation). I also wonder if we may want to only periodically log "Event size exceeds max_record_size_bytes" message (something similar to rate limited warn in the agent code) to avoid potential noise / flood of those messages in case we encounter many such events.

In addition to that, we may also want to add periodic DEBUG log with event content (pre-truncation) to make troubleshooting / similar easier.

Reply:
> Since previously we wouldn't perform additional processing in such case on the client side, right (aka we would send it as-is and server would reject / truncate it)?

Right. This has actually been weighing on my mind a lot. I don't actually know how often logstash handles records that exceed the max record size (whatever that value is), or the arbitrary value we chose of 200KiB.

How does the server deal with events that exceed the max record size? Our docs don't mention a record size limit, only a post body size limit.
@Kami How does the agent deal with large events? I couldn't find an answer to that question quickly on my own, and didn't want to delay this feature in order to dig around the agent codebase to find out.

Reply:
As mentioned on Slack, agent datasets are usually different - there is usually a single potentially large ``message`` attribute vs logstash scenario where it's common to have many other attributes (in addition or without the ``message`` attribute) on the event.

Having said that, I don't know how agent handles that so I will dig into code and also try to test it out.